pkg/auth/user/RoleGroupService.go

```diff
@@ -777,6 +783,146 @@ func (impl RoleGroupServiceImpl) DeleteRoleGroup(bean *bean.RoleGroup) (bool, er
+		return err
+	}
+	// delete mappings from orchestrator
+	err = impl.deleteMappingsFromOrchestrator(request.Ids, tx)
+	if err != nil {
+		impl.logger.Errorw("error in deleteRoleGroupsByIds", "request", request, "err", err)
+		return err
+	}
+	// update models to inactive with audit
+	err = impl.roleGroupRepository.UpdateToInactiveByIds(request.Ids, tx, request.LoggedInUserId)
+	if err != nil {
+		impl.logger.Errorw("error in deleteMappingsFromOrchestrator", "err", err)
+		return err
+	}
+
+	// delete from casbin
+	err = impl.deleteMappingsFromCasbin(groupCasbinNames, len(request.Ids))
+	if err != nil {
+		impl.logger.Errorw("error in deleteRoleGroupsByIds", "request", request, "err", err)
+		return err
+	}
+	// commit transaction
+	err = impl.roleGroupRepository.CommitATransaction(tx)
+	if err != nil {
+		impl.logger.Errorw("error in committing a transaction in deleteRoleGroupsByIds", "err", err)
+		return err
+	}
+	return nil
+
+}
+
+// deleteMappingsFromOrchestrator deletes role group role mapping from orchestrator only, takes in ids and returns error
+func (impl RoleGroupServiceImpl) deleteMappingsFromOrchestrator(roleGroupIds []int32, tx *pg.Tx) error {
+	roleGroupRoleMappings, err := impl.roleGroupRepository.GetRoleGroupRoleMappingIdsByGroupIds(roleGroupIds)
+	if err != nil {
+		impl.logger.Errorw("error in deleteMappingsFromOrchestrator", "err", err)
+		return err
+	}
+
+	mappingIds := make([]int, 0, len(roleGroupRoleMappings))
+	for _, model := range roleGroupRoleMappings {
+		mappingIds = append(mappingIds, model.Id)
+	}
+
+	if len(mappingIds) > 0 {
+		err = impl.roleGroupRepository.DeleteRoleGroupRoleMappingByIds(mappingIds, tx)
+		if err != nil {
+			impl.logger.Errorw("error in deleteMappingsFromOrchestrator", "err", err)
+			return err
+		}
+	}
+	return nil
+}
+
+// deleteMappingsFromCasbin delete GROUP-POLICY mappings and USER-GROUP mappings from casbin
+func (impl RoleGroupServiceImpl) deleteMappingsFromCasbin(groupCasbinNames []string, totalCount int) error {
+	groupNameVsCasbinRolesMap := make(map[string][]string, totalCount)
+	groupVsUsersMap := make(map[string][]string, totalCount)
+	for _, casbinName := range groupCasbinNames {
+		casbinRoles, err := casbin2.GetRolesForUser(casbinName)
+		if err != nil {
+			impl.logger.Warnw("No Roles Found for user", "casbinName", casbinName, "err", err)
+			return err
+		}
+		allUsersMappedToGroup, err := casbin2.GetUserByRole(casbinName)
+		if err != nil {
+			impl.logger.Errorw("error while fetching all users mapped to given group", "err", err)
+			return err
+		}
+		groupNameVsCasbinRolesMap[casbinName] = casbinRoles
+		groupVsUsersMap[casbinName] = allUsersMappedToGroup
+
+	}
+	// GROUP-POLICY mapping deletion from casbin
+	success := impl.userCommonService.DeleteRoleForUserFromCasbin(groupNameVsCasbinRolesMap)
+	if !success {
+		impl.logger.Errorw("error in deleteMappingsFromCasbin, not all mappings removed ", "success", success)
```

Comment:
success is always false, add payload in log instead

Reply:
DONE